### PR TITLE
SystemMonitor: Kill/stop UI improvements

### PR DESCRIPTION
**Pull request by sppmacd**
**SystemMonitor: Add Delete as alternate shortcut for killing a process**
    
This matches behaviour of other "system monitors", like Windows's Task
Manager

**SystemMonitor: Ask user before trying to kill/stop process**
    
These actions can be dangerous, especially for important system
processes. Let's make accidental breaking of the system a bit
harder. :^)

### CODE REVIEW

File: Userland/Applications/SystemMonitor/main.cpp [hunk 1]
```diff
@@ -254,18 +255,31 @@ int main(int argc, char** argv)
...
+    };
+
     auto kill_action = GUI::Action::create(
-        "&Kill Process", { Mod_Ctrl, Key_K }, Gfx::Bitmap::try_load_from_file("/res/icons/16x16/kill.png").release_value_but_fixme_should_propagate_errors(), [&](const GUI::Action&) {
+        "&Kill Process", { Mod_Ctrl, Key_K }, { Key_Delete }, Gfx::Bitmap::try_load_from_file("/res/icons/16x16/kill.png").release_value_but_fixme_should_propagate_errors(), [&](const GUI::Action&) {
             pid_t pid = selected_id(ProcessModel::Column::PID);
-            if (pid != -1)
+            if (pid == -1)
+                return;
+            auto rc = GUI::MessageBox::show(window, String::formatted("Do you want to kill \"{}\" (PID {})?", selected_name(ProcessModel::Column::Name), pid), "SystemMonitor", GUI::MessageBox::Type::Question, GUI::MessageBox::InputType::YesNo);
```
Comment (sw1tchbl4d3r) [inline Userland/Applications/SystemMonitor/main.cpp:270]:
```suggestion
            auto rc = GUI::MessageBox::show(window, String::formatted("Do you really want to kill \"{}\" (PID {})?", selected_name(ProcessModel::Column::Name), pid), "SystemMonitor", GUI::MessageBox::Type::Question, GUI::MessageBox::InputType::YesNo);
```
Nit: I would reword this to "Do you really want to kill", as this is a confirmation dialog.

File: Userland/Applications/SystemMonitor/main.cpp [hunk 2]
```diff
@@ -254,18 +255,31 @@ int main(int argc, char** argv)
...
         },
         &process_table_view);
 
     auto stop_action = GUI::Action::create(
         "&Stop Process", { Mod_Ctrl, Key_S }, Gfx::Bitmap::try_load_from_file("/res/icons/16x16/stop-hand.png").release_value_but_fixme_should_propagate_errors(), [&](const GUI::Action&) {
             pid_t pid = selected_id(ProcessModel::Column::PID);
-            if (pid != -1)
+            if (pid == -1)
+                return;
+            auto rc = GUI::MessageBox::show(window, String::formatted("Do you want to stop \"{}\" (PID {})?", selected_name(ProcessModel::Column::Name), pid), "SystemMonitor", GUI::MessageBox::Type::Question, GUI::MessageBox::InputType::YesNo);
```
Comment (sw1tchbl4d3r) [inline Userland/Applications/SystemMonitor/main.cpp:281]:
```suggestion
            auto rc = GUI::MessageBox::show(window, String::formatted("Do you really want to stop \"{}\" (PID {})?", selected_name(ProcessModel::Column::Name), pid), "SystemMonitor", GUI::MessageBox::Type::Question, GUI::MessageBox::InputType::YesNo);
```
Same as above.

**Review by sw1tchbl4d3r — COMMENTED** [review]
Just a tiny bit of polish.

File: Userland/Applications/SystemMonitor/main.cpp (hunk 1 above)
Comment (sppmacd) [reply Userland/Applications/SystemMonitor/main.cpp]: Fixed

File: Userland/Applications/SystemMonitor/main.cpp (hunk 2 above)
Comment (sppmacd) [reply Userland/Applications/SystemMonitor/main.cpp]: Fixed

File: Userland/Applications/SystemMonitor/main.cpp [hunk 3]
```diff
@@ -254,18 +255,31 @@ int main(int argc, char** argv)
...
+        return pid_index.data().to_string();
+    };
+
     auto kill_action = GUI::Action::create(
         "&Kill Process", { Mod_Ctrl, Key_K }, { Key_Delete }, Gfx::Bitmap::try_load_from_file("/res/icons/16x16/kill.png").release_value_but_fixme_should_propagate_errors(), [&](const GUI::Action&) {
             pid_t pid = selected_id(ProcessModel::Column::PID);
-            if (pid != -1)
+            if (pid == -1)
+                return;
+            auto rc = GUI::MessageBox::show(window, String::formatted("Do you really want to kill \"{}\" (PID {})?", selected_name(ProcessModel::Column::Name), pid), "SystemMonitor", GUI::MessageBox::Type::Question, GUI::MessageBox::InputType::YesNo);
```
Comment (linusg) [inline Userland/Applications/SystemMonitor/main.cpp:270]: While the program name itself is `FooBar`, visually we always stylize it as `Foo Bar` - in window titles, application shortcuts, etc.

```suggestion
            auto rc = GUI::MessageBox::show(window, String::formatted("Do you really want to kill \"{}\" (PID {})?", selected_name(ProcessModel::Column::Name), pid), "System Monitor", GUI::MessageBox::Type::Question, GUI::MessageBox::InputType::YesNo);
```

File: Userland/Applications/SystemMonitor/main.cpp [hunk 4]
```diff
@@ -254,18 +255,31 @@ int main(int argc, char** argv)
...
         },
         &process_table_view);
 
     auto stop_action = GUI::Action::create(
         "&Stop Process", { Mod_Ctrl, Key_S }, Gfx::Bitmap::try_load_from_file("/res/icons/16x16/stop-hand.png").release_value_but_fixme_should_propagate_errors(), [&](const GUI::Action&) {
             pid_t pid = selected_id(ProcessModel::Column::PID);
-            if (pid != -1)
+            if (pid == -1)
+                return;
+            auto rc = GUI::MessageBox::show(window, String::formatted("Do you really want to stop \"{}\" (PID {})?", selected_name(ProcessModel::Column::Name), pid), "SystemMonitor", GUI::MessageBox::Type::Question, GUI::MessageBox::InputType::YesNo);
```
Comment (linusg) [inline Userland/Applications/SystemMonitor/main.cpp:281]:
```suggestion
            auto rc = GUI::MessageBox::show(window, String::formatted("Do you really want to stop \"{}\" (PID {})?", selected_name(ProcessModel::Column::Name), pid), "System Monitor", GUI::MessageBox::Type::Question, GUI::MessageBox::InputType::YesNo);
```

**Review by linusg — CHANGES_REQUESTED** [review]

File: Userland/Applications/SystemMonitor/main.cpp (hunk 3 above)
Comment (sppmacd) [reply Userland/Applications/SystemMonitor/main.cpp]: Fixed

File: Userland/Applications/SystemMonitor/main.cpp (hunk 4 above)
Comment (sppmacd) [reply Userland/Applications/SystemMonitor/main.cpp]: Fixed

**Review by linusg — APPROVED** [review]
